src/executor/validate.rs

```diff
@@ -0,0 +1,204 @@
+use futures::stream::{self, TryStreamExt};
+use sqlparser::ast::{ColumnDef, ColumnOption, Ident};
+use std::{cmp::Ordering, fmt::Debug, rc::Rc};
+
+use crate::data::{Row, Schema, Value, ValueError};
+use crate::result::Result;
+use crate::store::Store;
+
+use super::execute::ExecuteError;
+
+pub enum ColumnValidation {
+    All,
+    SpecifiedColumns(Vec<Ident>),
+}
+
+pub async fn validate_rows<'a, T: 'static + Debug>(
+    storage: &impl Store<T>,
+    table_name: &str,
+    column_validation: ColumnValidation,
+    rows: &'a [Row],
+) -> Result<()> {
+    validate_rows_by(storage, table_name, column_validation, rows, |r| r).await
+}
+
+pub async fn validate_rows_by<'a, T: 'static + Debug, U: 'static + Debug, F>(
+    storage: &impl Store<T>,
+    table_name: &str,
+    column_validation: ColumnValidation,
+    items: &'a [U],
+    item_to_row: F,
```

Comment:
How about using `Iterator` than `&[U]` or `&[Row]`.
Then you don't need to use this kind of closure functions and you can merge `validate_rows` and `validate_rows_by` into a single function.

Here in the `validate_rows_by` function, `U` contains `(T, Row)` but `T` is not used in this function.
It means that `T` should be excluded outside of the function `validate_rows_by`, not inside.